packages/babel-plugin-transform-parameters/src/params.js

```diff
@@ -36,6 +36,10 @@ const iifeVisitor = {
       path.stop();
     }
   },
+  // type annotations don't use or introduce "real" bindings
+  "TypeAnnotation|TSTypeAnnotation"(path) {
```

Comment:
Ideally we may expose `bindingType` of `checkLVal` to `parseBindingList` so we do not register bindings of lval when they are inside a type annotation.

BTW we should also skip TypeParameters.
```ts
function a(b = <T extends (c: any) => void>(): void => {}) {
  let c;
}
```

Reply:
Since identifiers are the most often created/moved nodes, I think that it would be safer to add a `skip` like this one to `@babel/traverse` rather than relying on what was initially parsed.